source/data-modeling/field-types.txt

```diff
@@ -0,0 +1,558 @@
+
+The type of the ``preferences`` field is ``Object`` and will vary depending on
+the type of data stored in that field. The following example saves data into the
+``preferences`` field in two different ways:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-untyped
+   :end-before: # end-untyped
+
+Because {+odm+} doesn't perform any type conversions on untyped fields when
+reading from the database, values that require special handling might not work
+correctly in an untyped field. The following BSON data types should not be
+stored in untyped fields:
+
+- ``Date``: Returns as ``Time`` in untyped fields
+- ``DateTime``: Returns as ``Time`` in untyped fields
+- ``Range``: Returns as ``Hash`` in untyped fields
+
+StringifiedSymbol
+~~~~~~~~~~~~~~~~~
+
+Use the ``StringifiedSymbol`` field type to store values that should be exposed
+as symbols to {+language+} applications. ``StringifiedSymbol`` allows you to use symbols
+while ensuring interoperability with other drivers. This type stores all data on
+the database as strings, and converts the strings to symbols when read by the
+application. Values that cannot be directly converted to symbols, such as
+integers and arrays, are converted into strings and then into symbols.
+
+The following example stores a ``status`` field as a ``StringifiedSymbol`` and
+demonstrates how the field is stored and returned:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-stringified-symbol
+   :end-before: # end-stringified-symbol
+
+Hash
+~~~~
+
+You can store Hash data in a field by using the ``Hash`` type. When you specify
+a field as a ``Hash``, adhere to the MongoDB :manual:`Naming
+Restrictions </reference/limits/#naming-restrictions>` to ensure that the values
+store properly in the database.
+
+The following example creates a ``Person`` class and specifies the ``url`` field
+as a ``Hash``. 
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-hash
+   :end-before: # end-hash
+
+Time
+~~~~
+
+You can store values as ``Time`` instances by using the ``Time`` field value.
+``Time`` fields are stored in the time zone configured for your application. To
+learn more about configuring time zones, see the :ref:`Time Zones <time-zones>`
+guide. 
+
+.. TODO: Update Time Zones guide ref if it's changed during standardization
+
+The following example creates a ``Voter`` class and specifies the
+``registered_at`` property as a ``Time`` field:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-time
+   :end-before: # end-time
+
+.. note::
+
+   Storing a ``Date`` or ``DateTime`` value in a field specified as ``Time``
+   converts the value to ``Time`` when assigned. If you store a string in a
+   ``Time`` field, {+odm+} parses the string by using the ``Time.parse`` method.
+
```

Comment:
S: I think theres more explanation/example of this in the specify a query page

Reply:
Added as a todo